src/libs/actions/PersistedRequests.js

```diff
@@ -25,7 +25,15 @@ function save(requestsToPersist) {
  * @param {Object} requestToRemove
  */
 function remove(requestToRemove) {
-    persistedRequests = _.reject(persistedRequests, (persistedRequest) => _.isEqual(persistedRequest, requestToRemove));
+    /**
+     * Removes the first matching request, even if there are multiple matching requests.
+     * Because if multiple identical requests are queued. We want to keep the same requests then see more here: https://github.com/Expensify/App/issues/19640
```

Comment:
```suggestion
     * We only remove the first matching request because the order of requests matters. 
     * If we were to remove all matching requests, we can end up with a final state that is different than what the user intended.
```